docs/modules/ROOT/pages/design/cloudevents.adoc

```diff
@@ -0,0 +1,125 @@
+the following goals:
+
+* KubeArchive receives all Cloud Events from the Knative ApiServerSource
+* Cloud Events sent to KubeArchive can be replayed
+* Cloud Events received by KubeArchive, that cannot be processed successfully, are reported back to the sender as
+  delivery failure
+* Cloud Events sent to the sink will attempt to be redelivered if the initial delivery fails
+* Cloud Events that could not be delivered are replayed to KubeArchive automatically when KubeArchive becomes available
+  again
+
+== Design
+
+KubeArchive needs to use the functionality provided by Knative Eventing, Brokers and Triggers as well as features
+provided by the Cloud Events Specification to achieve an event driven architecture, where events are not lost and events
+can be replayed safely without worries about side effects.
+
+This diagram provides a high level overview of the life cycle of Cloud Events look like in KubeArchive and how the
+different components of the Cloud Event Delivery Pipeline interact with eachother.
+
+image::ce-pipeline.png[]
+
+=== Knative Eventing Brokers
+
+Instead of the ApiServerSource sending Cloud Events directly to the sink, the Cloud Events should instead be sent to a
+Knative Eventing Broker. Brokers can receive, persist, and send messages. Using the broker allows messages to be
+sent somewhere that they can be stored long term and is separate from the KubeArchive sink and Knative Eventing
+ApiServerSource, so if either are restarted, the Cloud Events won't be lost. KubeArchive will use two brokers. One
+primary broker (called broker) and secondary broker (called dead letter sink). The default configuration of
+Knative Eventing will have these brokers be a created as In Memory Brokers, but changing the Knative Eventing
+configuration can allow this broker to be created, for example, as a Kafka Broker instead
+
+[IMPORTANT]
+====
+For Production workloads, Knative Eventing strongly recommends using Brokers types other than the In Memory Broker.
+====
+
+==== Primary Broker
+
+The primary broker receives all of Cloud Events from the ApiServerSources created by the KubeArchive Operator.
+KubeArchive uses a trigger to subscribe to Cloud Events from this broker. The primary broker is configured as follows:
+
+[source,yaml]
+----
+spec:
+  delivery:
+    retry: 4
+    backoffPolicy: exponential
+    # ISO 8601 format
+    backoffDelay: PT0.5S
+    deadLetterSink:
+      ref:
+        apiVersion: eventing.knative.dev/v1
+        kind: Broker
+        name: kubearchive-dls
+----
+
+If the initial delivery fails, the broker will attempt to deliver the cloud event 4 additional times. When attempting to
+retry delivering the event, the broker will use an exponential backoff algorithm to calculate the amount of time to wait
+between retry attempts, and if all retry attempts fail, the broker will send the cloud event to the dead letter sink.
```

Comment:
```suggestion
If the initial delivery fails, the broker attempts to deliver the cloud event 4 additional times using an exponential backoff algorithm. If all retry attempts fail, the broker sends the Cloud Event to the dead letter sink broker.
```

Reply:
Why not just say?
```
If the initial delivery fails, the broker will try to resend the Cloud Event
using the the configured `retry`and `backoffPolicy` parameters.
These parameters may be changed by the user, see the Knative
documentation for details. If all retry attempts fail, the broker sends
the Cloud Event to the dead letter sink broker.
```
I don't see any good reason to include the actual default values in the text. What if they change?

Reply:
I made the change here and took the suggestion to not mention specific values but instead list the params that will affect if/how retries will be done